types/underscore/underscore-tests.ts

```diff
@@ -831,7 +867,7 @@ declare const extractChainTypes: ChainTypeExtractor;
     const collectionFunctionIteratee: _.Iteratee<_.Dictionary<StringRecord> | StringRecord[], string> = (element, key, collection) => {
         element; // $ExpectType StringRecord
         key; // $ExpectType string | number
-        collection; // $ExpectType Dictionary<StringRecord> | StringRecord[]
+        collection; // $ExpectType StringRecord[] | Dictionary<StringRecord>
```

Comment:
The linter is unfortunately not always the best at staying consistent with its exacting expectations when changes are made. Having needed to investigate further due to issues that I'm having on a difficult branch that are more difficult to resolve, this is due to the current lack of a type relationships API in typescript (https://github.com/microsoft/dtslint/issues/57#issuecomment-451666294 and https://github.com/microsoft/dtslint/issues/192 are the two issues I've found on this that help explain why this sort of thing occurs sometimes).

Reply:
On a sad note, due to the combination of quirks like these, an incomplete representation of JavaScript's semantics, "extensions" that end up conflicting with later standard JS features (e.g. class fields, decorators), and the need to rely on third-party typings that tend to require workarounds (Underscore has become exceptionally accurate thanks to your efforts), I feel nowadays that TS is causing more trouble overall than it is solving. While I'll happily keep helping to improve typings for libraries like Underscore and Backbone, I'm switching back to CoffeeScript for my own projects.

Reply:
Sorry to hear that; hopefully the situation will improve and you'll find it useful to use again one day 😄

Reply:
Also thanks for the compliment on my impact here! I agree that there are issues of varying severity with several typings, and after having gone through all the effort of updating the Underscore typings I will admit that it can take a surprising amount of work to make them accurate (and I haven't even covered all of Underscore yet, just the parts that I and my coworkers use most).